docs/book/developer-guide/advanced-usage/secret-references.md

```diff
@@ -0,0 +1,50 @@
+---
+description: How to reference secrets when configuring stack components
+---
+
+Some of the components in your stack require you to configure them with 
+sensitive information like passwords or tokens so they can connect to the 
+underlying infrastructure. Secret references allow you to configure these components in
+a secure way by not specifying the value directly but instead referencing a secret.
+To reference a secret in any string attribute of your stack components, simply specify
+the attribute value using the following syntax:
+```shell
+{{<SECRET_NAME>.<SECRET_KEY>}}
+```
+
+For example:
+```shell
+zenml metadata-store register secure_metadata_store \
+    --flavor=mysql \
+    --username='{{mysql_secret.username}}' \
+    --password='{{mysql_secret.password}}' \
+    ...
+```
+
+## Register missing secrets for your stack
+
+When using components with secret references in your stack, you need to make sure
+that the stack contains a [secrets manager](../../mlops-stacks/secrets-managers/secrets-managers.md)
+and all the referenced secrets exist in this secrets manager. To make this process easier, you can
+use the following CLI command to interactively register all secrets for a stack:
+```shell
+zenml stack register-secrets [<STACK_NAME>]
```

Comment:
Does this command check if all referenced secrets are present in the secrets manager? Isn't a similar check also performed when a stack is registered/updated?

Reply:
This indirectly makes sure that all the referenced secrets are present:
It checks all the secret references of the stack, and then queries the secrets manager:
- If the secret/key already exists, it asks if you want to update the secret or skip
- If the secret/key don't exists, it requires you to enter a value for it

That makes sure that once this command finishes, all secrets for the stack exist.

Reply:
The check when a stack is registered/updated instead does not interactively ask users to input secrets, as at that point the stack might not have a secrets manager.